streams/src/main/java/org/apache/kafka/streams/processor/internals/metrics/StreamsMetricsImpl.java

```diff
@@ -210,11 +211,13 @@ public final Sensor threadLevelSensor(final String threadId,
                                           final Sensor... parents) {
         final String key = threadSensorPrefix(threadId);
         synchronized (threadLevelSensors) {
-            threadLevelSensors.putIfAbsent(key, new LinkedList<>());
             final String fullSensorName = key + SENSOR_NAME_DELIMITER + sensorName;
-            final Sensor sensor = metrics.sensor(fullSensorName, recordingLevel, parents);
-            threadLevelSensors.get(key).push(fullSensorName);
-            return sensor;
+            return Optional.ofNullable(metrics.getSensor(fullSensorName))
+                    .orElseGet(() -> {
+                        threadLevelSensors.putIfAbsent(key, new LinkedList<>());
+                        threadLevelSensors.get(key).push(fullSensorName);
+                        return metrics.sensor(fullSensorName, recordingLevel, parents);
+                    });
```

Comment:
Although, you simplified this code, you did not apply all simplifications I proposed in PR #7914. I think you can even simplify this code further as shown here:
```
return Optional.ofNullable(metrics.getSensor(fullSensorName)).orElseGet(() -> {
    threadLevelSensors.computeIfAbsent(key, ignored -> new LinkedList<>()).push(fullSensorName);
    return metrics.sensor(fullSensorName, recordingLevel, parents);
});
```
This simplification can be applied also to the other methods below.

If you have any concerns about this simplifications please share your thoughts.

Reply:
oh good!!